_includes/v19.1/app/insecure/gorm-sample.go

```diff
@@ -0,0 +1,158 @@
+	const addr = "postgresql://maxroach@localhost:26257/bank?sslmode=disable"
+	db, err := gorm.Open("postgres", addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	// Set to `true` and GORM will print out all DB queries.
+	db.LogMode(false)
+
+	// Automatically create the "accounts" table based on the Account
+	// model.
+	db.AutoMigrate(&Account{})
+
+	// Insert two rows into the "accounts" table.
+	db.Create(&Account{ID: 1, Balance: 1000})
+	db.Create(&Account{ID: 2, Balance: 250})
+
+	// The sequence of steps in this section is:
+	// 1. Print account balances.
+	// 2. Set up some Accounts and transfer funds between them.
+	// 3. Print account balances again to verify the transfer occurred.
+
+	// Print balances before transfer.
+	printBalances(db)
+
+	var amount = 100
+	var fromAccount Account
+	var toAccount Account
+
+	db.First(&fromAccount, 1)
+	db.First(&toAccount, 2)
+
+	// Transfer funds between accounts.  To handle any possible
+	// transaction retry errors, we add a retry loop with exponential
+	// backoff to the transfer logic (see implementation below).
+	if err := transferFunds(db, fromAccount, toAccount, amount); err != nil {
+		// If the error is returned, it's either:
+		//   1. Not a transaction retry error, i.e., some other kind of
+		//   database error that you should handle here.
+		//   2. A transaction retry error that has occurred more than N
+		//   times (defined by the `maxRetries` variable inside
+		//   `transferFunds`), in which case you will need to figure out
+		//   why your database access is resulting in so much contention
+		//   (see 'Understanding and avoiding transaction contention':
+		//   https://www.cockroachlabs.com/docs/stable/performance-best-practices-overview.html#understanding-and-avoiding-transaction-contention)
+		fmt.Println(err)
+	}
+
+	// Print balances after transfer to ensure that it worked.
+	printBalances(db)
+
+	// Delete accounts so we can start fresh when we want to run this
+	// program again.
+	deleteAccounts(db)
+}
+
+func transferFunds(db *gorm.DB, fromAccount Account, toAccount Account, amount int) error {
+	if fromAccount.Balance < amount {
+		return fmt.Errorf("account %d balance %d is lower than transfer amount %d", fromAccount.ID, fromAccount.Balance, amount)
+	}
+
+	var maxRetries = 3
+	for retries := 0; retries <= maxRetries; retries++ {
+		if retries == maxRetries {
+			return fmt.Errorf("hit max of %d retries, aborting", retries)
+		}
+
+		txn := db.Begin()
+		// The statement below is used for forcing transaction retries
+		// for verifying the retry loop logic.  It is not necessary
+		// for production code.
+		// txn.Exec("SELECT now()")
+		if err := txn.Exec(
+			// The below statement is used for forcing transaction
+			// retries for verifying the retry loop logic.  It is not
+			// necessary for production code.
+			// `SELECT crdb_internal.force_retry('1s'::INTERVAL)`,
+			`UPSERT INTO accounts (id, balance) VALUES
```

Comment:
Users of an ORM don't generally want to write SQL queries by hand like this. Idiomatic use of GORM would be to either modify the Account objects and call Save on them or use gorm's Update method.

Reply:
Updated to modify the Account objects and call Save on them.